lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -569,6 +570,31 @@ sub prepare_job_results {
     return (\%archs, \%results, $aggregated);
 }
 
+# appends the specified $distri and $version to $array_to_add_parts_to as string or if $raw as Mojo::ByteStream
+sub _add_distri_and_version_to_summary {
+    my ($array_to_add_parts_to, $distri, $version, $raw) = @_;
+
+    for my $part ($distri, $version) {
+        # handle case when multiple distri/version parameters have been specified
+        $part = $part->{-in} if (ref $part eq 'HASH');
+        next unless $part;
+
+        # separate distri and version with a whitespace
+        push(@$array_to_add_parts_to, ' ') if (@$array_to_add_parts_to);
+
+        if (ref $part eq 'ARRAY') {
+            # separate multiple distris/versions using a slash
+            if (@$part) {
+                push(@$array_to_add_parts_to, map { ($raw ? Mojo::ByteStream->new($_) : $_, '/') } @$part);
+                pop(@$array_to_add_parts_to);
```

Comment:
why push+pop one after another?

Reply:
To omit the last (and otherwise surplus) `/` separator. Maybe there is a better way to write it but I preferred this over a for loop with special handling for the first or last element.

Reply:
Does this explanation satisfy you? :-)

Reply:
ok.